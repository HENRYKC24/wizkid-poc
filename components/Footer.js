import React from "react";
import { Image, StyleSheet, Text, TouchableOpacity, View } from "react-native";
import handDollar from "../assets/images/hand_dollar.png";
import handPeople from "../assets/images/hand_people.png";
import lightBulb from "../assets/images/light_bulb.png";
import people from "../assets/images/people.png";

const Footer = () => {
  return (
    <View style={styles.container}>
      <TouchableOpacity>
        <View style={styles.group}>
          <Image style={styles.handDollar} source={handDollar} />
          <Text style={styles.baseText}>Earn</Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity>
        <View style={styles.group}>
          <Image style={styles.people} source={people} />
          <Text style={[styles.baseText, styles.share]}>Share</Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity>
        <View style={styles.group}>
          <Image style={styles.lightBulb} source={lightBulb} />
          <Text style={styles.baseText}>Learn</Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity>
        <View style={styles.group}>
          <Image style={styles.handPeople} source={handPeople} />
          <Text style={styles.baseText}>Rewards</Text>
        </View>
      </TouchableOpacity>
    </View>
  );
};

export default Footer;

const styles = StyleSheet.create({
  container: {
    position: "absolute",
    // flex: 1,
    backgroundColor: "#73a1d1",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    width: "100%",
    bottom: 0,
    margin: 0,
    paddingHorizontal: 42,
    height: 97.94,
  },
  group: {
    justifyContent: "center",
    alignItems: "center",
  },
  baseText: {
    color: "#000",
    fontSize: 12,
    fontWeight: "500",
  },
  share: {
    marginBottom: 12,
  },
});
