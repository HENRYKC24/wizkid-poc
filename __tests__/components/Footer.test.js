import React from 'react';
import renderer from 'react-test-renderer';

import Footer from '../../components/Footer';

describe('<Footer />', () => {
  it('has 4 children', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree.children.length).toBe(4);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});