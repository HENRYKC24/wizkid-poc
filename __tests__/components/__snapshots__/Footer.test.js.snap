// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#73a1d1",
      "bottom": 0,
      "flexDirection": "row",
      "height": 97.94,
      "justifyContent": "space-between",
      "margin": 0,
      "paddingHorizontal": 42,
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
      />
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 12,
            "fontWeight": "500",
          }
        }
      >
        Earn
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 12,
              "fontWeight": "500",
            },
            Object {
              "marginBottom": 12,
            },
          ]
        }
      >
        Share
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
      />
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 12,
            "fontWeight": "500",
          }
        }
      >
        Learn
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
      />
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 12,
            "fontWeight": "500",
          }
        }
      >
        Rewards
      </Text>
    </View>
  </View>
</View>
`;
